lib/trivial.nix

```
@@ -448,6 +448,39 @@ rec {
   isFunction = f: builtins.isFunction f ||
     (f ? __functor && isFunction (f.__functor f));
 
+  /* Copy the function arguments (__functionArgs) of the first function
+     onto the second function.
+
+     Type:
+       copyFunctionArgs :: ({ <a_i> } -> b) -> (a -> c) -> (({ <a_i> } -> c) & AttrSet)
```

```suggestion
       copyFunctionArgs :: (a -> b) -> (a -> c) -> (a -> c)
```

The `__functionArgs` and `__functor` stuff is fairly internal, we shouldn't try to document it.